### DISCUSSION TITLE
Deployment Preview Screen Out Issue

### DISCUSSION THREAD

**Discussion by johnjang94** [community/community#75219] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

Hello everyone,

I have a project that I have created using Vite, TypeScript, and React, and I am attempting a continuous deployment as I build the project. However, I am facing preview deployment failure issue with one of my repositories. This particular file where the error has occurred is about user login using email credentials or Google Authentication.
I have tried to deploy not only from here on Github Pages but also Vercel and Netlify.
The production deployment has been successful in all three platforms, but the preview deployment has not been successful no matter how I try to debug.

I have been able to learn further about the issue that is causing the preview of the deployment from Netlify and the detail is shown as follows:

```
12:27:45 AM: Netlify Build                                                 
12:27:45 AM: ────────────────────────────────────────────────────────────────
12:27:45 AM: ​
12:27:45 AM: ❯ Version
12:27:45 AM:   @netlify/build 29.26.3
12:27:45 AM: ​
12:27:45 AM: ❯ Flags
12:27:45 AM:   baseRelDir: true
12:27:45 AM:   buildId: 654f10be676e2a6c37d87ce6
12:27:45 AM:   deployId: 654f10be676e2a6c37d87ce8
12:27:45 AM: ​
12:27:45 AM: ❯ Current directory
12:27:45 AM:   /opt/build/repo
12:27:45 AM: ​
12:27:45 AM: ❯ Config file
12:27:45 AM:   No config file was defined: using default values.
12:27:45 AM: ​
12:27:45 AM: ❯ Context
12:27:45 AM:   deploy-preview
12:27:45 AM: ​
12:27:45 AM: Build command from Netlify app                                
12:27:45 AM: ────────────────────────────────────────────────────────────────
12:27:45 AM: ​
12:27:45 AM: $ npm run build
12:27:45 AM: > experiment2@0.0.0 build
12:27:45 AM: > tsc && vite build
12:27:47 AM: Failed during stage "building site": Build script returned non-zero exit code: 2
12:27:47 AM: src/contexts/AuthContext.tsx(3,3): error TS6133: "browserSessionPersistence" is declared but its value is never read.
12:27:47 AM: src/contexts/AuthContext.tsx(6,3): error TS6133: "setPersistence" is declared but its value is never read.
12:27:47 AM: ​
12:27:47 AM: "build.command" failed                                        
12:27:47 AM: ────────────────────────────────────────────────────────────────
12:27:47 AM: ​
12:27:47 AM:   Error message
12:27:47 AM:   Command failed with exit code 2: npm run build (https://ntl.fyi/exit-code-2)
12:27:47 AM: ​
12:27:47 AM:   Error location
12:27:47 AM:   In Build command from Netlify app:
12:27:47 AM:   npm run build
12:27:47 AM: ​
12:27:47 AM:   Resolved config
12:27:47 AM:   build:
12:27:47 AM:     command: npm run build
12:27:47 AM:     commandOrigin: ui
12:27:47 AM:     environment:
12:27:47 AM:       - REVIEW_ID
12:27:47 AM:     publish: /opt/build/repo/dist
12:27:47 AM:     publishOrigin: ui
12:27:47 AM: Build failed due to a user error: Build script returned non-zero exit code: 2
12:27:47 AM: Failing build: Failed to build site
12:27:47 AM: Finished processing build request in 19.839s
```

I understand that the data is from Netlify, so I am presuming that some of you are probably thinking "why are you asking your question here? This information is from Netlify, so you should reach out to Netlify community or Netlify customer service instead."

Yes, I have reached out to Netlify customer service or Netlify community to find out a solution for this.
In the meantime, I have the same preview issue when I deploy on Github Pages. This tells me that this is not the error caused by Netlify. Therefore, I need to find out what's happening.

The following information is my codebase:
```tsx

import {
  GoogleAuthProvider,
  createUserWithEmailAndPassword,
  onAuthStateChanged,
  signInWithEmailAndPassword,
  signInWithPopup,
  signOut,
} from 'firebase/auth'
import { createContext, useContext, useEffect, useReducer } from 'react'
import { auth } from '../components/api/firebase/index'
import useStatusMessages from '../hooks/StatusHook'

type TestState = {
  email: string
  password: string
  userInfo: any
}

type TestAction = {
  payload?: any
  type: string
}

type StatusStateType = {
  error: string
  success: string
}

type Cell = {
  AuthState: TestState
  AuthDispatch: React.Dispatch<TestAction>
  auth: any
  // auth
  googleAuth: () => void
  Register: () => void
  LogIn: () => void
  logOut: () => void
  //status
  setSuccess: (message: string) => void
  setError: (message: string) => void
  statusState: StatusStateType
}

const AuthContext = createContext<Cell | null>(null)

export const AuthContextProvider = ({
  children,
}: {
  children: React.ReactNode
}) => {
  //  status handling
  const { statusState, setError, setSuccess } = useStatusMessages({
    error: '',
    success: '',
  })
  const initialState: TestState = {
    email: '',
    password: '',
    userInfo: {},
  }

  const AuthReducer = (state: TestState, action: TestAction) => {
    switch (action.type) {
      case 'SET_EMAIL':
        return { ...state, email: action.payload }
      case 'SET_PASSWORD':
        return { ...state, password: action.payload }
      case 'SET_USER':
        return { ...state, userInfo: action.payload }

      default:
        return state
    }
  }
  const [AuthState, AuthDispatch] = useReducer(AuthReducer, initialState)
  //  after page reload we are getting user information from session and updating the global state of AuthState.userInfo with it
  useEffect(() => {
    onAuthStateChanged(auth, (currentUser) => {
      AuthDispatch({ type: 'SET_USER', payload: currentUser })
    })
  }, [])

  // authentication functionality
  // google auth

  const provider = new GoogleAuthProvider()
  //   this authentication functions would be better to be inside of the each respective component instad of context API

  const googleAuth = async () => {
    try {
      const user = await signInWithPopup(auth, provider)
      AuthDispatch({ type: 'SET_USER', payload: user.user })
      setSuccess('User Has Been Authenticated')
      console.log(user)
    } catch (error) {
      const err: any = error
      console.log(err.code)
      setError(err.code)
    }
  }

  // register
  const Register = async () => {
    const { email, password } = AuthState
    try {
      await createUserWithEmailAndPassword(auth, email, password)
      onAuthStateChanged(auth, (currentUser) => {
        AuthDispatch({ type: 'SET_USER', payload: currentUser })
      })
      setSuccess('User Has Been Created')
    } catch (error) {
      const err: any = error
      console.log(err.code)
      setError(err.code)
    }
  }
  // login
  const LogIn = async () => {
    const { email, password } = AuthState
    try {
      await signInWithEmailAndPassword(auth, email, password)
      onAuthStateChanged(auth, (currentUser) => {
        AuthDispatch({ type: 'SET_USER', payload: currentUser })
        console.log(currentUser)
      })
      setSuccess('User Has Logged In')
    } catch (error) {
      const err: any = error
      console.log(err.code)
      setError(err.code)
    }
  }
  // logout
  const logOut = async () => {
    await signOut(auth)
    AuthDispatch({ type: 'SET_USER', payload: {} })
    window.location.reload()
  }

  return (
    <AuthContext.Provider
      value={{
        AuthState,
        AuthDispatch,
        auth,
        logOut,
        Register,
        LogIn,
        googleAuth,
        statusState,
        setError,
        setSuccess,
      }}
    >
      {children}
    </AuthContext.Provider>
  )
}

export const UseAuthContext = () => {
  const context = useContext(AuthContext)
  if (!context) {
    throw new Error('Context Not Wrapped Reload Page')
  }

  return context
}
```

From the above code, you can tell that I actually do not have those codes that are reportedly to be the cause of the deployment failure. I want to find out what's then causing the issue and how I can debug this problem. I do not know if it can help you, but here is the specific URL address where this file belongs in my Github repository: [Authentication](https://github.com/johnjang94/project_ruminate/blob/front-end/src/contexts/AuthContext.tsx)

I would appreciate any suggestion.

thank you.

**Comment by sky9262** [comment]
Are you getting the same errors in your local?

**Reply by johnjang94** [reply]
Hi, 
thanks for your comment.
No, i do not get the preview error like this when i run it on my local server.
It happens when i deploy using Github Pages.

**Reply by sky9262** [reply]
I don't even get any preview of the website after running `npm run preview`

Just blank page!!!!!!

**Comment by glaizalei** [comment]
great, you doesn`t have any error!

**Comment by johnjang94** [comment]
Hi everyone,

I am here to leave an update regarding this case.
As you may notice from my original code, I used Netlify for deployment.
What I have learnt was that there was a line of code that caused the error on this deployment because it was not a part of code that was being used in the file. I have removed the line of code that caused the error and I thought that my action will fix the problem.

The problem was fixed as anticipated, but I was still being reported with the same error only for the preview mode.
If you have ever used Netlify deployment, you might get to know that Netlify has "preview" feature besides the production deployment.
Basically, the production deployment was successful after I removed the line of code that caused the problem. However, the preview mode was displaying as failure because it was keep rendering the original version that contained the error line of code rather than a new version that corrected the problem.

After some time, I have figured out how to use the preview mode properly.
Now, the codebase is fixed and the deployment is successful.

Thanks for all your attention.
